checks/dangerous_workflow.go

```diff
@@ -120,23 +128,70 @@ func validateGitHubActionWorkflowPatterns(path string, content []byte, dl checke
 		return false, err
 	}
 
+	// 3. Check for secrets used in workflows triggered by pull requests.
+	if err := validateSecretsInPullRequests(workflow, path, dl, pdata); err != nil {
+		return false, err
+	}
+
 	// TODO: Check other dangerous patterns.
 	return true, nil
 }
 
+func validateSecretsInPullRequests(workflow *actionlint.Workflow, path string,
+	dl checker.DetailLogger, pdata *patternCbData) error {
+	// We need pull request trigger.
+	if !usesPullRequestTrigger(workflow) {
+		return nil
+	}
+
+	// We need code checkout.
+	if !workflowUsesCodeCheckout(workflow) {
+		return nil
+	}
+
+	// Secrets used in env at the top of the wokflow.
+	if err := checkSecretInEnv(workflow.Env, path, dl, pdata); err != nil {
+		return err
+	}
+
+	// Secrets used on jobs.
+	for _, job := range workflow.Jobs {
+		if err := checkJobForUsedSecrets(job, path, dl, pdata); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
 func validateUntrustedCodeCheckout(workflow *actionlint.Workflow, path string,
 	dl checker.DetailLogger, pdata *patternCbData) error {
-	if checkPullRequestTrigger(workflow) {
-		for _, job := range workflow.Jobs {
-			if err := checkJobForUntrustedCodeCheckout(job, path, dl, pdata); err != nil {
-				return err
-			}
+	if !usesPullRequestTargetTrigger(workflow) {
+		return nil
+	}
+
+	for _, job := range workflow.Jobs {
+		if err := checkJobForUntrustedCodeCheckout(job, path, dl, pdata); err != nil {
+			return err
 		}
 	}
+
 	return nil
 }
 
-func checkPullRequestTrigger(workflow *actionlint.Workflow) bool {
+func usesPullRequestTrigger(workflow *actionlint.Workflow) bool {
+	// Check if the webhook event trigger is a pull_request_target
+	for _, event := range workflow.On {
+		e, ok := event.(*actionlint.WebhookEvent)
+		if ok && e.Hook != nil && e.Hook.Value == "pull_request" {
+			return true
+		}
+	}
+
+	return false
+}
+
+func usesPullRequestTargetTrigger(workflow *actionlint.Workflow) bool {
 	// Check if the webhook event trigger is a pull_request_target
 	for _, event := range workflow.On {
 		e, ok := event.(*actionlint.WebhookEvent)
```

Comment:
Same comment as above - can we replace with `event.EventName()`?

Reply:
I've replaced.